corpora-api.yml

```diff
@@ -0,0 +1,530 @@
+        a filter, then only projects that meet the status criteria will be outputted.
+      parameters:
+        - $ref: '#/components/parameters/query_user_uuid'
+        - name: from_date
+          in: query
+          schema:
+            type: integer
+          description: The date after which projects should have been created.
+        - name: to_date
+          in: query
+          description: The date before which projects should have been created.
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  request_id:
+                    type: string
+                  project_uuid:
+                    $ref: '#/components/schemas/project_uuid'
+                  from_date:
+                    type: integer
+                  to_date:
+                    type: integer
+        '400':
+          $ref: '#/components/responses/400'
+
+
+  /v1/projects/{project_uuid}:
+
+    get:
+      tags:
+        - projects
+      summary: Get Project
+      description: >-
+        This will return all datasets and associated attributes, including processing pipeline statuses of the project.
+      parameters:
+        - $ref: '#/components/parameters/path_project_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+    delete:
+      tags:
+        - projects
+      summary: Delete Project
+      description: >-
+        Deletes an entire project from Corpora, including any generated artifacts/assets and deployments. If
+        no such project exists, then a warning will be outputted.
+      parameters:
+        - $ref: '#/components/parameters/path_project_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/dataset/{dataset_uuid}:
+
+    post:
+      tags:
+        - dataset
+      summary: Request to download a dataset
+      description: >-
+        Request to download a file which on success will generate a pre-signed URL to download the dataset.
+      parameters:
+        - $ref: '#/components/parameters/path_dataset_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+components:
+
+  schemas:
+    user_uuid:
```

Comment:
This was likely addressed during @maniarathi's high level design review, but we'll need a User API to implement and enforce authn/z in the future.